macro/vignettes/design_meta_timestep.Rmd

```diff
@@ -0,0 +1,403 @@
+---
+title: "Design of Generation of Discrete Time Steps"
+output:
+  pdf_document: default
+  html_notebook: default
+---
+From `r system("git config --get remote.origin.url", intern = TRUE)`
+on `r date()`, generated by `r Sys.info()[["effective_user"]]`.
+
+```{r load_libraries}
+library(macro)
+```
+
+
+# Overview
+
+I'd like to be able to build a time step quickly from a specification, at least for some
+simple models. In this case, I'm thinking about specifying linear equations with coefficients
+that are constant across a time step.
+$$
+\frac{dx}{dt}=Ax=\left[\begin{array}{cc}-h & \sin(r) \\ h & -\sin(r)\end{array}\right]x
+$$
+I'm using a $\sin$ there to make the point that the rates can be arbitrary functions of
+parameters. We could write that equation as
+```{r tease_apart}
+si_equations <- list(
+U ~ -h * U + sin(r) * X,
+X ~ h * U - sin(r) * X
+)
+```
+I'll assume this is a differential form of an ODE that reflects a master equation.
+That is, I'll assume that everything that leaves one compartment enters another compartment
+and that there are no constant terms.
+
+**Are these limitations the best choices?**
+
+From this kind of specification, we can produce lots of different kinds of time steps.
```

Comment:
I'm not sure how this sentence answers/relates to the question raised in the bold text.